### PR TITLE
Verify the src exists in webdav MOVE

### LINKED ISSUE #20069: WebDAV MOVE on a non existing file results in Internal Server Error
### Steps to reproduce
1. In the web UI, create a folder named ```src``` in the root folder; leave it empty.
2. From command line, type
```curl -v -u [user]:[password] https://[domain_and_base_path]/remote.php/webdav/src/nofile.txt -X MOVE -H "Destination:https://[domain_and_base_path]/remote.php/webdav/nofile.txt" -H "Overwrite:F"```

### Expected behaviour
HTTP/1.1 404 Not Found
...
Exception with message ```File with name src/nofile.txt could not be located``` (seen in version 8.1.3)

### Actual behaviour
HTTP/1.1 500 Internal Server Error
...
No response body

### Server configuration
**Operating system**:
Ubuntu

**Web server:**
Apache/2.4.7

**Database:**
MySQL

**PHP version:**
PHP/5.5.9-1ubuntu4.13

**ownCloud version:** (see ownCloud admin page)
ownCloud 8.2.0 (stable)
8.2.0.12 in status.php

**Updated from an older ownCloud or fresh install:**
Fresh

**List of activated apps:**
Activity  2.1.3
Deleted files  0.7.0
File Locking  
First Run Wizard  1.1
Gallery  14.2.0
Mail Template Editor  0.1
Notifications  0.1.0
PDF Viewer  0.7
Provisioning API  0.3.0
Share Files  0.7.0
Text Editor  2.0
Updater  0.6
Versions  1.1.0
Video Viewer  0.1.3

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:**
No

**Are you using an external user-backend, if yes which one:**
No

#### ownCloud log (data/owncloud.log)
```Error	PHP	Call to a member function isDeletable() on a non-object at /opt/owncloud/lib/private/connector/sabre/filesplugin.php#135	2015-10-26T19:48:30+00:00 ```

#### EXTRA
There is no error if nofile.txt is "located" in the root folder, instead of a subfolder.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes https://github.com/owncloud/core/issues/20069

Unit test added.

@davivel please verify

CC: @DeepDiver1975 @PVince81 @MorrisJobke @nickvergessen

### CODE REVIEW

File: apps/dav/lib/connector/sabre/filesplugin.php [hunk 1]
```diff
@@ -132,6 +132,10 @@ function checkMove($source, $destination) {
 		if ($sourceDir !== $destinationDir) {
 			$sourceFileInfo = $this->fileView->getFileInfo($source);
 
+			if ($sourceFileInfo === false) {
+				throw new \Sabre\DAV\Exception\PreconditionFailed($source . ' does not exist');
```
Comment (nickvergessen) [inline apps/dav/lib/connector/sabre/filesplugin.php]: `Precondition`? I'd say `NotFound`.

**Comment by nickvergessen** [comment]
Will add a smashbox test for this, when it is possible.

**Comment by rullzer** [comment]
@nickvergessen thanks!

File: apps/dav/lib/connector/sabre/filesplugin.php (hunk 1 above)
Comment (rullzer) [inline apps/dav/lib/connector/sabre/filesplugin.php]: fair enough... let me fix that

Comment (rullzer) [inline apps/dav/lib/connector/sabre/filesplugin.php]: Fixed

**Comment by MorrisJobke** [comment]
Code makes sense and it fixes the issue :+1:

**Comment by MorrisJobke** [comment]
@karlitschek The issue was scheduled for 8.2.1. Can you please approve the backport? Thanks :)

**Comment by nickvergessen** [comment]
:+1:

**Comment by nickvergessen** [comment]
Smashbox test for coverage https://github.com/owncloud/smashbox/pull/131

**Comment by DeepDiver1975** [comment]
great :+1:

**Comment by DeepDiver1975** [comment]
@rullzer please prepare backport pr

**Comment by rullzer** [comment]
backport stable8 in https://github.com/owncloud/core/pull/20077

**Comment by DeepDiver1975** [comment]
> backport stable8 in #20077

THX

**Comment by davivel** [comment]
@rullzer , works for me too. Thanks.

**Comment by karlitschek** [comment]
Yes. backport makes sense . :+1: thanks

**Pull request merged**